sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -155,11 +169,19 @@ object ScalaReflection extends ScalaReflection {
     val walkedTypePath = WalkedTypePath().recordRoot(enc.clsTag.runtimeClass.getName)
     // Assumes we are deserializing the first column of a row.
     val input = GetColumnByOrdinal(0, enc.dataType)
-    val deserializer = deserializerFor(
-      enc,
-      upCastToExpectedType(input, enc.dataType, walkedTypePath),
-      walkedTypePath)
-    expressionWithNullSafety(deserializer, enc.nullable, walkedTypePath)
+    enc match {
+      case RowEncoder(fields) =>
```

Comment:
what encoder do we create for inner struct? how is it different from the root `RowEncoder`?

Reply:
We create one with null checks. This one does not need them because we always return a Row.